### PR TITLE
Add helper functions to bpf_endian.h

### PR DESCRIPTION
**Pull request by b1tg**
## Description

Add helper functions to `bpf_endian.h`, as discussed in issue https://github.com/microsoft/ebpf-for-windows/issues/1240

## Testing

No

## Documentation
No

### CODE REVIEW

**Comment by b1tg** [comment]
@microsoft-github-policy-service agree

**Review by dthaler — APPROVED** [review]
Thanks for this contribution!

**Comment by b1tg** [comment]
CI failed due to network issue, how can i re-run this?

**Comment by dthaler** [comment]
> CI failed due to network issue, how can i re-run this?

CI currently shows as passing for this PR so should be good.

File: include/bpf_endian.h [hunk 1]
```diff
@@ -4,13 +4,78 @@
 
 #include <stdint.h>
 
-inline __attribute__((always_inline)) uint16_t
-bpf_ntohs(uint16_t us)
-{
-    return us << 8 | us >> 8;
-}
+/*
```
Comment (dthaler) [inline include/bpf_endian.h:7]: It was just pointed out to me that it looks like this PR copies code from https://github.com/torvalds/linux/blob/master/tools/lib/bpf/bpf_endian.h
To adhere to both this project's Contributor License Agreement and the license in that other file:
if code was copied from there, it's important to acknowledge the source and the SPDX identifier of the source.
Hence assuming this was copied, please add a comment in the code acknowledging the source URL and citing
its SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause).

**Review by dthaler — CHANGES_REQUESTED** [review]

File: include/bpf_endian.h (hunk 1 above)
Comment (b1tg) [reply include/bpf_endian.h:11]: Does my new commit meet the requirements?

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**